### PR TITLE
F1 fix + log loss +  evaluate prequential delayed regression

### PR DESCRIPTION
**Pull request by danielnowakassis**
### F1 Score fix :

* deals with classes yet without predictions, which result in zero division. Now it maches sklearn. Also makes a difference between macro and micro F1
* pushes ROCAUC to BasicPerformanceEvaluator 

### Log loss :
* Adds the log loss metric, and maches sklearn

### evaluate prequential delayed regression

* Add prequential delayed evaluation for regression

### CODE REVIEW

**Comment by hmgomes** [comment]
thanks, already reviewed as part of the moa-capymoa branch, will proceed to merge
